packages/ember-metal/lib/property_get.js

```diff
@@ -95,56 +95,51 @@ export function get(obj, keyName) {
   let isFunction = type === 'function';
   let isObjectLike = isObject || isFunction;
 
-  let descriptor;
-  let value;
+  if (!isObjectLike) {
+    return obj[keyName];
+  }
 
-  if (isObjectLike) {
-    if (EMBER_METAL_TRACKED_PROPERTIES) {
-      let tracker = getCurrentTracker();
-      if (tracker) tracker.add(tagForProperty(obj, keyName));
-    }
+  if (EMBER_METAL_TRACKED_PROPERTIES) {
+    let tracker = getCurrentTracker();
+    if (tracker) tracker.add(tagForProperty(obj, keyName));
+  }
 
-    descriptor = descriptorFor(obj, keyName);
-
-    if (descriptor === undefined) {
-      if (DEBUG && HAS_NATIVE_PROXY) {
-        value = getPossibleMandatoryProxyValue(obj, keyName);
-      } else {
-        value = obj[keyName];
-      }
-
-      if (isDescriptor(value)) {
-        deprecate(
-          `[DEPRECATED] computed property '${keyName}' was not set on object '${obj &&
-            obj.toString &&
-            obj.toString()}' via 'defineProperty'`,
-          false,
-          {
-            id: 'ember-meta.descriptor-on-object',
-            until: '3.5.0',
-            url:
-              'https://emberjs.com/deprecations/v3.x#toc_use-defineProperty-to-define-computed-properties',
-          }
-        );
-        descriptor = value;
-      }
+  let descriptor = descriptorFor(obj, keyName);
+  let value;
+  if (descriptor === undefined) {
+    if (DEBUG && HAS_NATIVE_PROXY) {
+      value = getPossibleMandatoryProxyValue(obj, keyName);
+    } else {
+      value = obj[keyName];
     }
 
-    if (descriptor !== undefined) {
-      return descriptor.get(obj, keyName);
+    if (isDescriptor(value)) {
```

Comment:
it is weird to do this call, when at this point descriptor could be defined and value === undefined

Reply:
maybe add test case what you mean or suggest change :P ? I just reorganized `if`s but the logic is same as was before